cypress/integration/6-crypto/crypto.spec.ts

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/// <reference types="cypress" />
+
+import { MatrixClient, RoomStateEvent } from "matrix-js-sdk/src/matrix";
+
+import { SynapseInstance } from "../../plugins/synapsedocker";
+
+function waitForRoomEncryption(cli: MatrixClient, roomId: string, callback: () => void) {
+    // @ts-ignore - protected
+    if (cli.roomList.isRoomEncrypted(roomId)) {
+        callback();
+    } else {
+        cli.once(RoomStateEvent.Update, () => waitForRoomEncryption(cli, roomId, callback));
```

Comment:
You'll need to use the `RoomStateEvent` from the window as `js-sdk` isn't available at runtime in the CI

Reply:
See https://github.com/matrix-org/matrix-react-sdk/blob/develop/docs/cypress.md#using-matrix-js-sdk

Reply:
Good catch, thank you!